authors/apps/authentication/views.py

```diff
@@ -69,3 +78,77 @@ def update(self, request, *args, **kwargs):
         serializer.save()
 
         return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+class PasswordReset(CreateAPIView):
+    permission_classes = (AllowAny,)
+    # renderer_classes = (UserJSONRenderer,)
+    serializer_class = EmailCheckSerializer
+
+    def post(self, request):
+        email = request.data.get('email', {})
+        serializer = self.serializer_class(data={"email": email})
+        serializer.is_valid(raise_exception=True)
+        try:
+            user = User.objects.get(email=email)
+            username = user.username
+        except User.DoesNotExist:
+            message = {"message": "The Email provided is not registered"}
+            return Response(message, status=status.HTTP_400_BAD_REQUEST)
+
+        token = jwt.encode({
+            "email": email, "iat": datetime.now(),
+            "exp": datetime.utcnow() + timedelta(seconds=5)},
```

Comment:
You set expiry to 5secs
No wonder the link expired before I could even finish typing the password
Kindly add some more time

Reply:
Just updated that to 24 hours expiration time. Kindly check